src/components/DataDimension/ItemOptionsSelector/ItemOptionsSelector.js

```diff
@@ -0,0 +1,302 @@
+import { Button, IconArrowLeft16, Transfer } from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React, { useEffect, useState } from 'react'
+import { apiFetchOptions } from '../../../api/dimensions.js'
+import i18n from '../../../locales/index.js'
+import {
+    DIMENSION_TYPE_PROGRAM_DATA_ELEMENT,
+    DIMENSION_TYPE_PROGRAM_ATTRIBUTE_OPTION,
+    DIMENSION_TYPE_PROGRAM_DATA_ELEMENT_OPTION,
+    DIMENSION_TYPE_PROGRAM_ATTRIBUTE,
+} from '../../../modules/dataTypes.js'
+import {
+    getIcon,
+    getDimensionType,
+} from '../../../modules/dimensionListItem.js'
+import {
+    TRANSFER_HEIGHT,
+    TRANSFER_OPTIONS_WIDTH,
+    TRANSFER_SELECTED_WIDTH,
+} from '../../../modules/dimensionSelectorHelper.js'
+import { useDebounce, useDidUpdateEffect } from '../../../modules/utils.js'
+import styles from '../../styles/DimensionSelector.style.js'
+import { SelectedEmptyPlaceholder } from '../SelectedEmptyPlaceholder.js'
+import { SourceEmptyPlaceholder } from '../SourceEmptyPlaceholder.js'
+import { TransferOption } from '../TransferOption.js'
+import itemOptionSelectorStyles from './styles/ItemOptionSelector.style.js'
+
+const formatOptionsFilters = (dataItemType, dataItemId) => {
+    const optionsFilters = { dataItemId }
+
+    if (dataItemType === DIMENSION_TYPE_PROGRAM_DATA_ELEMENT) {
+        optionsFilters.dataType = DIMENSION_TYPE_PROGRAM_DATA_ELEMENT_OPTION
+    } else if (dataItemType === DIMENSION_TYPE_PROGRAM_ATTRIBUTE) {
+        optionsFilters.dataType = DIMENSION_TYPE_PROGRAM_ATTRIBUTE_OPTION
+    }
+
+    return optionsFilters
+}
+
+export const ItemOptionsSelector = ({
+    id: dataItemId,
+    name: dataItemName,
+    type: dataItemType,
+
+    selectedItems,
+    infoDataItem,
+    setInfoDataItem,
+    displayNameProp,
+    itemsRef,
+    onEditClick,
+    onSelect,
+    onClose,
+    dataTest,
+}) => {
+    const [state, setState] = useState({
+        searchTerm: '',
+        filter: formatOptionsFilters(dataItemType, dataItemId),
+        options: [],
+        loading: true,
+        nextPage: 1,
+    })
+
+    const debouncedSearchTerm = useDebounce(state.searchTerm, 500)
+    const dataEngine = useDataEngine()
+
+    const setSearchTerm = (searchTerm) =>
+        setState((state) => ({ ...state, searchTerm }))
+
+    const fetchItems = async (page) => {
+        setState((state) => ({
+            ...state,
+            nextPage: page === 1 ? 1 : state.nextPage,
+            loading: true,
+        }))
+        const result = await apiFetchOptions({
+            dataEngine,
+            nameProp: displayNameProp,
+            page,
+            filter: state.filter,
+            searchTerm: state.searchTerm,
+        })
+        const newOptions = []
+        result.dimensionItems?.forEach((item) => {
+            newOptions.push({
+                label: item.name,
+                value: item.id,
+                disabled: item.disabled, // XXX is this returned by the api?!
```

Comment:
Is it okay if this is `undefined` ?

Reply:
Most of the code here is copied from `ItemSelector`.
I guess it's fine if it's `undefined`, but my comment there was because I wasn't sure where this can be initialised.